result/result.go

```diff
@@ -161,61 +165,71 @@ func (r *Results) normalizeTracerouteHops() {
 }
 
 func (r *Results) normalizeE2eProbe() {
-	r.E2eProbe.RTTs = []float64{}
-	var packetSent, packetReceived int
-	var totalRTTs, minRTT, maxRTT float64
-	RTTs := []float64{}
+	r.E2eProbe.PacketsSent = len(r.E2eProbe.RTTs)
 
-	// TODO: Replace with "50x e2e probe impl"
-	//       Right now, we temporarily use single Traceroute data to fill e2e probe
-	for _, run := range r.Traceroute.Runs {
-		packetSent++
-		destHop := run.getDestinationHop()
-		if destHop == nil {
-			continue
-		}
+	// Count received packets (non-zero RTTs) and collect valid RTTs
+	var validRTTs []float64
+	packetsReceived := 0
 
-		packetReceived++
-		if destHop.RTT > maxRTT {
-			maxRTT = destHop.RTT
-		}
-		if destHop.RTT < minRTT || minRTT == 0 {
-			minRTT = destHop.RTT
+	for _, rtt := range r.E2eProbe.RTTs {
+		if rtt > 0.0 {
+			packetsReceived++
+			validRTTs = append(validRTTs, rtt)
 		}
-		RTTs = append(RTTs, destHop.RTT)
-
-		totalRTTs += destHop.RTT
-	}
-
-	if packetReceived > 0 {
-		r.E2eProbe.RTT.Avg = totalRTTs / float64(packetReceived)
-	}
-	r.E2eProbe.RTT.Min = minRTT
-	r.E2eProbe.RTT.Max = maxRTT
-	r.E2eProbe.PacketsSent = packetSent
-	r.E2eProbe.PacketsReceived = packetReceived
-	r.E2eProbe.PacketLossPercentage = float32(packetSent-packetReceived) / float32(packetSent)
-	r.E2eProbe.RTTs = RTTs
-
-	// compute jitter
-	// TODO: this is one way of computing jitter (https://www.rfc-editor.org/rfc/rfc4689.html#section-3.2.5)
-	//       but actual jitter computation for e2e probe will require further investigation
-	//       to choose the most suitable way to compute jitter.
-	if len(RTTs) >= 2 {
-		var totalJitter float64
-		var prevRtt float64
-		for _, rtt := range RTTs {
-			if prevRtt == 0 {
-				prevRtt = rtt
-				continue
+	}
+
+	r.E2eProbe.PacketsReceived = packetsReceived
+
+	// Calculate packet loss percentage
+	if r.E2eProbe.PacketsSent > 0 {
+		r.E2eProbe.PacketLossPercentage = float32(r.E2eProbe.PacketsSent-r.E2eProbe.PacketsReceived) / float32(r.E2eProbe.PacketsSent)
+	}
+
+	if len(validRTTs) > 0 {
+		var totalRTTs float64
+		minRTT := validRTTs[0]
+		maxRTT := validRTTs[0]
+
+		for _, rtt := range validRTTs {
+			totalRTTs += rtt
+			if rtt < minRTT {
+				minRTT = rtt
+			}
+			if rtt > maxRTT {
+				maxRTT = rtt
 			}
-			totalJitter += math.Abs(rtt - prevRtt)
 		}
-		r.E2eProbe.Jitter = totalJitter / float64(len(RTTs)-1)
+
+		r.E2eProbe.RTT.Avg = totalRTTs / float64(len(validRTTs))
+		r.E2eProbe.RTT.Min = minRTT
+		r.E2eProbe.RTT.Max = maxRTT
+	} else {
+		// No packets received
+		r.E2eProbe.RTT.Avg = 0.0
+		r.E2eProbe.RTT.Min = 0.0
+		r.E2eProbe.RTT.Max = 0.0
+	}
```

Comment:
```suggestion
	}
```
this code is not needed right? the values should be already 0.0

Reply:
Yup.